### PR TITLE
A lot of tiny cleanups to exceptions

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libdevcore/IpfsHash.cpp [hunk 1]
```diff
@@ -55,11 +55,7 @@ string base58Encode(bytes const& _data)
 
 bytes dev::ipfsHash(string _data)
 {
-	if (_data.length() >= 1024 * 256)
-		BOOST_THROW_EXCEPTION(
-			DataTooLong() <<
-			errinfo_comment("Ipfs hash for large (chunked) files not yet implemented.")
-		);
+	assertThorw(_data.length() < 1024 * 256, DataTooLong, "IPFS hash for large (chunked) files not yet implemented.");
```
Comment (chriseth) [inline libdevcore/IpfsHash.cpp]: We should implement that soon.

**Review by chriseth — DISMISSED** [review]

**Comment by leonardoalt** [comment]
Tests failing

**Comment by axic** [comment]
Actually adding tests for `toHex`/`fromHex` before we merge this.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Ah, I still haven't pushed the tests, waiting for soltest to compile.

**Pull request merged**